engine/consensus/compliance/engine.go

```diff
@@ -293,7 +322,25 @@ func (e *Engine) processAvailableMessages() error {
 	for {
 		// TODO prioritization
 		// eg: msg := engine.SelectNextMessage()
```

Comment:
I think the following code already has the desired prioritization. 
But we don't check for a termination signal here.